taichi/transforms/demote_operations.cpp

```diff
@@ -118,8 +118,8 @@ class DemoteOperations : public BasicStmtVisitor {
                is_signed(lhs->element_type())) {
       // @ti.func
       // def bit_shr(a, b):
-      //     signed_a = ti.cast(a, ti.uXX)
-      //     shifted = ti.bit_sar(signed_a, b)
+      //     unsigned_a = ti.cast(a, ti.uXX)
+      //     shifted = ti.bit_sar(unsigned_a, b)
```

Comment:
Why is it sar here?

Reply:
`sar` for unsigned is equivalent to `shr`.

Reply:
I guess the main goal of the demotion is to make codegen simpler (only `sar` needed).